### PR TITLE
A bit of an authority leak: dapp-svelte-wallet purses exposed to dapps via pursesNotifier.

### PR DESCRIPTION
**Pull request by zarutian**
The Proplem: purses of dapp-svelte-wallet are exposed to dapps via `E(walletBridge).getPursesNotifier()` as its implementation exposed the pursesFullState to dapps instead of just pursesJSONState.

An unscrupulus dapp could there for do
```javascript
const steal = (walletBridge) => {
  const board = E(walletBridge).getBoard();
  const sack = E(board).getValue("«sackBoardId»");
  const PN = E(walletBridge).getPursesNotifier();
  return E(PN).getUpdateSince().then({ value: pursesFullState } => E(sack).putValuables(pursesFullState.map(pfs => pfs.purse)));
}
```
or even worse if the next last line is replaced with `return E(sack).robForever(PN);`

Have not written a test for this yet.
Paging @katelynsills , @michaelfig , @erights , and anyone who wants to take a look as this is a bit serious.

### CODE REVIEW

**Comment by zarutian** [comment]
btw wanted to ask: any reason why issuersNotifier of the wallet is not exposed to dapps? I surmise that issuers are not meant to be closesly held.

I ask only because I was scrutenizing these parts in detail.

**Comment by katelynsills** [comment]
Great catch @zarutian! I can confirm this is a bug. I've added myself as a reviewer. Let me know when you think it's ready for review. Btw, what is this notation meant to do: `(depositBoardId && { depositBoardId })`?

**Comment by zarutian** [comment]
saw `...(depositBoardId && { depositBoardId })` at line 237 (start of jstate object literal). It uses the spread operator on the result of that ?mis?-use of the and operator to conditionally put in depositBoardId, as far as I can make out. The condition being that depositBoardId is not falsy.

I suspect it is too clever by half as indicated by your question.``

**Comment by zarutian** [comment]
Hmm... this is mostly ready for review except it is missing tests that tests for this leakage. I am not skilled enough test writer yet to write such a test in such a way that it detects this leakage in all cases. So I apreciate all help there.

**Comment by katelynsills** [comment]
> btw wanted to ask: any reason why issuersNotifier of the wallet is not exposed to dapps? I surmise that issuers are not meant to be closesly held.
> 
> I ask only because I was scrutenizing these parts in detail.

I think it could be (should be?) exposed to dapps. We're still figuring out the exact API.

**Comment by katelynsills** [comment]
> saw `...(depositBoardId && { depositBoardId })` at line 237 (start of jstate object literal). It uses the spread operator on the result of that ?mis?-use of the and operator to conditionally put in depositBoardId, as far as I can make out. The condition being that depositBoardId is not falsy.
> 
> I suspect it is too clever by half as indicated by your question.``

Gotcha, thanks. I see now that it was already present. Feel free to keep it in or simplify.

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js [hunk 1]
```diff
@@ -160,6 +160,45 @@ export function makeWallet({
   const { notifier: pursesNotifier, updater: pursesUpdater } = makeNotifierKit(
     [],
   );
+  /** @type {NotifierRecord<PursesJSONState[]>} */
+  const {
+    notifier: attenuatedPursesNotifier,
+    updater: attenuatedPursesUpdater,
+  } = makeNotifierKit([]);
+  {
+    // explict whitelist
```
Comment (katelynsills) [inline packages/dapp-svelte-wallet/api/src/lib-wallet.js]:
```suggestion
    // explicit whitelist
```

**Review by katelynsills — COMMENTED** [review]
Looks good to me! One tiny change in a comment. I will make a PR to test this, and then at that point, I will approve this PR and we can merge this.

**Comment by katelynsills** [comment]
Here's the PR that adds tests! https://github.com/Agoric/agoric-sdk/pull/2226/

**Comment by zarutian** [comment]
Curious, pursesNotifier and attenuatedPursesNotifier should produce same data at the same time. But it might be that observeIteration lags behind by one event-loop turn. Investigating.

**Comment by zarutian** [comment]
@katelynsills: ready for review.

**Review by katelynsills — APPROVED** [review]
LGTM! Thank you for finding and fixing this authority leak.

**Comment by zarutian** [comment]
You are welcome.

**Pull request merged**